interpreter/text/parser.mly

```diff
@@ -574,34 +580,44 @@ elemref :
   | LPAR REF_NULL RPAR { let at = at () in fun c -> ref_null @@ at }
   | LPAR REF_FUNC var RPAR { let at = at () in fun c -> ref_func ($3 c func) @@ at }
 
-passive_elemref_list :
+elemref_list :
   | /* empty */ { fun c -> [] }
-  | elemref passive_elemref_list { fun c -> $1 c :: $2 c }
+  | elemref elemref_list { fun c -> $1 c :: $2 c }
 
-active_elemref_list :
+elemindex_list :
   | var_list
     { let f = function {at; _} as x -> ref_func x @@ at in
       fun c lookup -> List.map f ($1 c lookup) }
 
 elem :
-  | LPAR ELEM bind_var_opt elem_type passive_elemref_list RPAR
+  | LPAR ELEM bind_var_opt offset elemindex_list RPAR  /* Sugar */
     { let at = at () in
       fun c -> ignore ($3 c anon_elem bind_elem);
-      fun () -> Passive {etype = $4; data = $5 c} @@ at }
-  | LPAR ELEM bind_var var offset active_elemref_list RPAR
-    { let at = at () in
-      fun c -> ignore (bind_elem c $3);
       fun () ->
-      Active {index = $4 c table; offset = $5 c; init = $6 c func} @@ at }
-  | LPAR ELEM var offset active_elemref_list RPAR
+        ActiveWithIndices {index = 0l @@ at; offset = $4 c; init = $5 c func} @@ at }
+  | LPAR ELEM bind_var_opt extern_kind elemindex_list RPAR
```

Comment:
Is it possible to factor out the choice between `extern_kind elemindex_list` and `elem_type elemref_list` into a separate production, thereby avoid all the multiplicative cases here?

Reply:
Yeah, it's a it annoying, but it follows from the independence of the flag bits in the flag word, see [here](https://github.com/WebAssembly/bulk-memory-operations/issues/98#issuecomment-507330729).

For the text format, I think we can factorise the legacy support slightly differently and do what you're asking, i.e., only allow omitting the elem_kind. Then you should be able to do the refactoring I suggested with only one additional special case for legacy active.